src/lib/snarky/src/libsnark.ml

```diff
@@ -149,6 +153,15 @@ struct
         let z = stub x y in
         schedule_delete z ; z
 
+    (* XXX: if add changes to not allocate, this is a problem *)
+    let dup x = add zero x
```

Comment:
Probably worth adding an explicit `dup` in the C-wrapper which just does an allocation rather than relying on implementation detail of `add`